tests/performance_tests/performance_test_basic.py

```diff
@@ -15,218 +15,291 @@
 from TestHarness.TestHelper import AppArgs
 import log_reader
 
-Print = Utils.Print
-errorExit = Utils.errorExit
-cmdError = Utils.cmdError
-relaunchTimeout = 30
-emptyBlockGoal = 5
-
-def fileOpenMode(filePath) -> str:
+class PerformanceBasicTest():
+    @dataclass
+    class TestHelperConfig():
+        killAll: bool = True # clean_run
+        dontKill: bool = False # leave_running
+        keepLogs: bool = False
+        dumpErrorDetails: bool = False
+        delay: int = 1
+        nodesFile: str = None
+        verbose: bool = False
+        _killEosInstances: bool = True
+        _killWallet: bool = True
+
+        def __post_init__(self):
+            self._killEosInstances = not self.dontKill
+            self._killWallet = not self.dontKill
+
+    @dataclass
+    class ClusterConfig():
+        pnodes: int = 1
+        totalNodes: int = 2
+        topo: str = "mesh"
+        extraNodeosArgs: str = ' --http-max-response-time-ms 990000 --disable-subjective-api-billing true '
+        useBiosBootFile: bool = False
+        genesisPath: str = "tests/performance_tests/genesis.json"
+        maximumP2pPerHost: int = 5000
+        maximumClients: int = 0
+        loggingDict = { "bios": "off" }
+        _totalNodes: int = 2
+
+        def __post_init__(self):
+            self._totalNodes = max(2, self.pnodes if self.totalNodes < self.pnodes else self.totalNodes)
+
+    def __init__(self, testHelperConfig: TestHelperConfig=TestHelperConfig(), clusterConfig: ClusterConfig=ClusterConfig(), targetTps: int=8000, testTrxGenDurationSec: int=30, tpsLimitPerGenerator: int=4000, numAddlBlocksToPrune: int=2, rootLogDir: str=os.path.splitext(os.path.basename(__file__))[0], saveJsonReport: bool=False):
```

Comment:
There comes a point when the code is begging for a carriage return...

Reply:
Never!  There comes a point when the code is begging for a bigger monitor.  Point taken.  I hadn't done any sort of final formatting sweep.  I'll take care of this.  Thanks.

Reply:
Addressed: [11179c7](https://github.com/AntelopeIO/leap/pull/302/commits/11179c764269316ce52d8f6d099530d805a0d9d4)

Reply:
I bought my first UltraHD monitor in 2014.  :stuck_out_tongue:  I am aware that other people still don't have one, and Github intentionally doesn't insert linewraps the way a sophisticated editor does, so I do appreciate a few carriage returns.